src/commands/NFD.ts

```diff
@@ -0,0 +1,1003 @@
+    for (let i = 0; i < 2; i++) {
+      const nfd = await this.getNFDByName(nfdNames[i])
+      // Confirm the NFD exists
+      if (!nfd) {
+        return interaction.reply({ content: `I couldn't find an NFD with the name "${nfdNames[i]}".`, ephemeral: true })
+      }
+
+      // Confirm that the caller owns the NFD
+      if (nfd.owner != interaction.user.id) {
+        return interaction.reply({
+          content: `You don't own "${nfdNames[i]}"! You can't slurp something you don't own!`,
+          ephemeral: true,
+        })
+      }
+    }
+
+    // Both NFDs are verified now. First generate a new NFD before deleting anything.
+    const parts = await this.makeNFDcode()
+
+    // Check to see if we failed to make a unique one
+    if (!parts) {
+      return interaction.reply({
+        content: "I tried really hard but I wasn't able to make a unique NFD for you. Sorry... :'(",
+        ephemeral: true,
+      })
+    }
+
+    // We have the new NFD ready to go. Delete the old two.
+    this.client.nFDItem
+      .deleteMany({
+        where: {
+          name: {
+            in: [first, second],
+          },
+        },
+      })
+      .then(() => {
+        this.composeNFD(parts)
+      })
+      .then(() => {
+        return this.storeNFDinDatabase(parts, getCallerFromCommand(interaction))
+      })
+      .then((nfd) => {
+        this.makeReply(nfd, interaction, ownerMember)
+      })
+      .then(() => {
+        this.updateDBSuccessfulSlurp(ownerMember.id)
+      })
+      .catch((err) => {
+        interaction.reply({ content: 'The dinochain broke... what a surprise', ephemeral: true }).catch((err) => {
+          console.error('Something really went wrong minting this NFD...', err)
+        })
+      })
+  }
+
+  private getParts(): BodyParts {
+    const imageList = fs.readdirSync(this.FRAGMENT_PATH)
+    const bodyList = imageList.filter((filename) => filename.includes('_b.png'))
+    const mouthList = imageList.filter((filename) => filename.includes('_m.png'))
+    const eyesList = imageList.filter((filename) => filename.includes('_e.png'))
+
+    const body = getRandomElement(bodyList)
+    const mouth = getRandomElement(mouthList)
+    const eyes = getRandomElement(eyesList)
+
+    const code = `${body},${mouth},${eyes}`
+
+    return { body: body, mouth: mouth, eyes: eyes, code: code }
+  }
+
+  private async composeNFD(parts: BodyParts) {
+    const out = sharp(path.join(this.FRAGMENT_PATH, parts.body)).composite([
+      {
+        input: path.join(this.FRAGMENT_PATH, parts.mouth),
+        blend: 'over',
+      },
+      {
+        input: path.join(this.FRAGMENT_PATH, parts.eyes),
+        blend: 'over',
+      },
+    ])
+    if (!parts.fileName) {
+      parts.fileName = parts.name + '.png'
+    }
+    await out.toFile(path.join(this.OUTPUT_PATH, parts.fileName))
+    return Promise.resolve(parts)
+  }
+
+  private async getNFDByCode(code: string) {
+    return await this.client.nFDItem.findUnique({
```

Comment:
`return await` in an async function is redundant. The await gets you the actual value but the return in an async function gets wrapped back into a Promise.

Reply:
Good to know. Removed this anti-pattern throughout